### PR TITLE
StartEpochSealingBuffer triggers packing on timer

### PR DESCRIPTION
**Pull request by ZenGround0**
## Proposed Changes
Improvements to snap deals integration beyond the existing mvp PR
- [x] use the config's `StartEpochSealingBuffer` param as a way to enforce that sectors are packed for sealing / updating no matter how many deals they have if the nearest deal start date is close enough to the present.  This solves an existing problem predating snap deals (FYI @jennijuju)
- [x] Config option for disabling automatic creation of new sectors for handling incoming deals.  Forces user to trigger inclusion by marking sectors for snap upgrade.
~- [ ] Recycle assigned deals in updating sectors that have been aborted so that these deals can be reassigned to other sectors~

### CODE REVIEW

**Comment by jennijuju** [comment]
Nice - I assume it is intentional that this will apply to regular deal sectors too?

**Comment by jennijuju** [comment]
Please do **not ** merge this PR without a pr to https://lotus.filecoin.io/docs/storage-providers/config/

**Comment by ZenGround0** [comment]
It turns out that recycling deals from a failed sector is a more involved change than I had thought and therefore is going to have to wait beyond v 1.15.0.  I am going to file an issue which I'll link to from back here describing the problem a bit more.

File: node/config/types.go [hunk 1]
```diff
@@ -115,6 +115,8 @@ type DealmakingConfig struct {
 	// This includes the time the deal will need to get transferred and published
 	// before being assigned to a sector
 	ExpectedSealDuration Duration
+	// Whether new sectors are sealed to pack incoming deals
```
Comment (magik6k) [inline node/config/types.go:118]:
```suggestion
	// Whether new sectors are created to pack incoming deals
```

Comment (magik6k) [reply node/config/types.go:118]: Possibly worth elaborating what happens when this is set to false, and why one would do that

File: extern/storage-sealing/input.go [hunk 2]
```diff
@@ -117,9 +118,29 @@ func (m *Sealing) maybeStartSealing(ctx statemachine.Context, sector SectorInfo,
 			return false, xerrors.Errorf("getting storage config: %w", err)
 		}
 
-		// todo check deal age, start sealing if any deal has less than X (configurable) to start deadline
 		sealTime := time.Unix(sector.CreationTime, 0).Add(cfg.WaitDealsDelay)
 
+		// check deal age, start sealing when the deal closest to starting is within slack time
+		safeSealTime := sealTime
```
Comment (magik6k) [inline extern/storage-sealing/input.go:124]: I don't think we need this copied variable, can just use `sealTime` directly

**Review by magik6k — COMMENTED** [review]
Looks good, just 2 small nits

**Comment by BlocksOnAChain** [comment]
@ZenGround0 can we merge it for 1.15.0, since I think we got positive feedback from Magik?
CC: @jennijuju

**Review by arajasek — APPROVED** [review]

**Comment by jennijuju** [comment]
oh btw! bad pr title!! reminder to follow the one in pr template next time

**Comment by ZenGround0** [comment]
@jennijuju You'll need to help me read the template better because I tried my best to follow it.

> <PR type>: <area>: <change being made>

The title is feat: sealing: <change being made> right?

> `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_

feat

> `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_

sealing

Are the underscores what you are talking about?

**Comment by jennijuju** [comment]
what you said is all ✅  

when I left the comment, the title was " StartEpochSealingBuffer triggers packing on timer " and I update it!

**Pull request merged**